application/src/main/java/org/togetherjava/tjbot/commands/moderation/ReportCommand.java

```diff
@@ -188,9 +189,13 @@ private MessageCreateAction createModMessage(String reportReason,
                             "Go to Message", guild.getId(), reportedMessage.channelID,
                             reportedMessage.id));
 
-        Optional<Role> moderatorRole =
-                guild.getRolesByName(configModGroupPattern, true).stream().findFirst();
-        moderatorRole.ifPresent(role -> message.setContent(role.getAsMention()));
+        Optional<String> roleName = guild.getRoles()
+            .stream()
+            .map(Role::getName)
+            .filter(configModGroupPattern)
+            .findFirst();
+        roleName.ifPresent(
+                s -> message.setContent(guild.getRolesByName(s, false).get(0).getAsMention()));
```

Comment:
I think it would be a bit easier to read if you continue with the optional api for one more step:
```java
Optional<Role> moderatorRole = ...
.filter(...)
.findFirst()
.map(name -> guild.getRolesByName(name, false).get(0));
```